onnxruntime/core/providers/shared/node_unit/node_unit.h

```diff
@@ -64,17 +64,26 @@ class NodeUnit {
   ProviderType GetExecutionProviderType() const noexcept;
 
   const Node& GetNode() const noexcept { return target_node_; }
-  const std::vector<const Node*> GetOutputNodes() const noexcept { return output_nodes_; }
+  const std::vector<const Node*>& GetOutputNodes() const noexcept { return output_nodes_; }
 
  private:
   std::vector<NodeUnitIODef> inputs_;
   std::vector<NodeUnitIODef> outputs_;
 
   const std::vector<const Node*> output_nodes_;  // all the nodes producing outputs for this NodeUnit
   const Node& target_node_;
-  Type type_;
+  const Type type_;
 
-  void InitForNode();  // Initializing for single Node
+  // Initializing for a single Node
+  void InitForSingleNode();
+
+  // Initializing for a QDQ group
+  void InitForQDQGroup(const QDQ::NodeGroup& node_group);
 };
 
+// Get all the nodes in the given graph_viewer as NodeUnits (SingleNode or QDQGroup)
+// And return a map to quick query the NodeUnit which contains the given Node
```

Comment:
perhaps explicitly say that the values of the map point to NodeUnits owned by the vector?